ui/guest/src/components/Guest.js

```diff
@@ -1007,14 +1101,34 @@ export function Guest(props) {
       if (isNullOrUndefined(record)) {
         throw new Error('No record found for dispatcher element');
       }
-
       handler(event, record);
 
     } else {
       return true;
     }
   }
 
+  function scrollToReceptacle(receptacles) {
+    let elementInView;
+    let element;
+    elementInView = forEach(receptacles, ({ id }) => {
+      let elem = ElementRegistry.fromICEId(id).element;
+      if (isElementInView(elem)) {
+        elementInView = true;
+        element = elem;
+        return 'break';
+      }
+    }, false);
+
+    if (!elementInView) {
+      // TODO: Do this relative to the scroll position. Don't move if things are already in viewport. Be smarter.
+      let element = ElementRegistry.fromICEId(receptacles[0].id).element;
+      $(scrollElement).animate({
+        scrollTop: $(element).offset().top - 100
+      }, 300);
+    }
+  }
```

Comment:
Is the bug we saw fixed?

Reply:
no, it is just the function scrollToReceptacle with changes to use the receptacles object